rtgui/gradient.cc

```diff
@@ -579,6 +580,19 @@ bool Gradient::drag1(int modifierKey)
     return false;
 }
 
+void Gradient::releaseEdit()
+{
+    if (lastObject >= 0) {
```

Comment:
It's hard to tell from the diff alone, if those `object` comparisons (`>= 0`, `> 1`, `> 0`) are correct, if "invalid" has different representations. Maybe that could be better modelled with a `std::optional<>` (which is C++17, but I have a simple C++11 implementation to share).

Reply:
`lastObject` gets its value from `EditDataProvider::getObject()` and I don't feel like changing that class or working around it in this PR.